File: Parker/src/app/models/user.model.ts
export class User {
  public id: string;
  public type: string;
  public lastName: string;
  public firstName: string;
  public email: string;
  public phone: string;
  public password: string;
  public avatar: string;

  constructor(
    id?: string,
    type?: string,
    lastName?: string,
    firstName?: string,
    email?: string,
    phone?: string,
    password?: string,
    avatar?: string
  ) {
    this.id = id || this.id;
    this.type = type || this.type;
    this.lastName = lastName || this.lastName;
    this.firstName = firstName || this.firstName;
    this.email = email || this.email;
    this.phone = phone || this.phone;
    this.password = password || this.password;
    this.avatar = avatar || this.avatar;
  }
}

File: server/OcticonApi/OcticonApi.Core/Models/User/UserModel.cs
using OcticonApi.Core.Domain.Repository;
using OcticonApi.Utils;
using System;

namespace OcticonApi.Core.Models.User
{
    public class UserModel
    {
        public Int32 Id { get; set; }
        public UserType Type { get; set; }
        public String LastName { get; set; }
        public String FirstName { get; set; }
        public String Email { get; set; }
        public String Phone { get; set; }

        public static UserModel FromDbModel(UserInfo user)
        {
            return new UserModel
            {
                Id = user.Id,
                FirstName = user.FirstName,
                LastName = user.LastName,
                Email = user.Email,
                Phone = user.Phone,
                Type = (UserType)Enum.Parse(typeof(UserType), user.Type, true)
            };
        }
    }
}
